tests/e2e/e2e_setup_test.go

```diff
@@ -0,0 +1,418 @@
+			},
+		},
+	})
+
+	bz, err := cdc.MarshalJSON(&bankGenState)
+	s.Require().NoError(err)
+	appGenState[banktypes.ModuleName] = bz
+
+	var genUtilGenState genutiltypes.GenesisState
+	s.Require().NoError(cdc.UnmarshalJSON(appGenState[genutiltypes.ModuleName], &genUtilGenState))
+
+	// generate genesis txs
+	genTxs := make([]json.RawMessage, len(c.validators))
+	for i, val := range c.validators {
+		createValmsg, err := val.buildCreateValidatorMsg(stakeAmountCoin)
+		s.Require().NoError(err)
+
+		signedTx, err := val.signMsg(createValmsg)
+		s.Require().NoError(err)
+
+		txRaw, err := cdc.MarshalJSON(signedTx)
+		s.Require().NoError(err)
+
+		genTxs[i] = txRaw
+	}
+
+	genUtilGenState.GenTxs = genTxs
+
+	bz, err = cdc.MarshalJSON(&genUtilGenState)
+	s.Require().NoError(err)
+	appGenState[genutiltypes.ModuleName] = bz
+
+	bz, err = json.MarshalIndent(appGenState, "", "  ")
+	s.Require().NoError(err)
+
+	genDoc.AppState = bz
+
+	bz, err = tmjson.MarshalIndent(genDoc, "", "  ")
+	s.Require().NoError(err)
+
+	// write the updated genesis file to each validator
+	for _, val := range c.validators {
+		writeFile(filepath.Join(val.configDir(), "config", "genesis.json"), bz)
+	}
+}
+
+func (s *IntegrationTestSuite) initValidatorConfigs(c *chain) {
+	for i, val := range c.validators {
+		tmCfgPath := filepath.Join(val.configDir(), "config", "config.toml")
+
+		vpr := viper.New()
+		vpr.SetConfigFile(tmCfgPath)
+		s.Require().NoError(vpr.ReadInConfig())
+
+		valConfig := &tmconfig.Config{}
+		s.Require().NoError(vpr.Unmarshal(valConfig))
+
+		valConfig.P2P.ListenAddress = "tcp://0.0.0.0:26656"
+		valConfig.P2P.AddrBookStrict = false
+		valConfig.P2P.ExternalAddress = fmt.Sprintf("%s:%d", val.instanceName(), 26656)
+		valConfig.RPC.ListenAddress = "tcp://0.0.0.0:26657"
+		valConfig.StateSync.Enable = false
+		valConfig.LogLevel = "info"
+
+		var peers []string
+
+		for j := 0; j < len(c.validators); j++ {
+			if i == j {
+				continue
+			}
+
+			peer := c.validators[j]
+			peerID := fmt.Sprintf("%s@%s%d:26656", peer.nodeKey.ID(), peer.moniker, j)
+			peers = append(peers, peerID)
+		}
+
+		valConfig.P2P.PersistentPeers = strings.Join(peers, ",")
+
+		tmconfig.WriteConfigFile(tmCfgPath, valConfig)
+
+		// set application configuration
+		appCfgPath := filepath.Join(val.configDir(), "config", "app.toml")
+
+		appConfig := srvconfig.DefaultConfig()
+		appConfig.API.Enable = true
+		appConfig.MinGasPrices = fmt.Sprintf("%s%s", minGasPrice, photonDenom)
+
+		srvconfig.WriteConfigFile(appCfgPath, appConfig)
+	}
+}
+
+func (s *IntegrationTestSuite) runValidators(c *chain) {
+	s.T().Logf("starting Gaia %s validator containers...", c.id)
+
+	s.valResources[c.id] = make([]*dockertest.Resource, len(c.validators))
+	for i, val := range c.validators {
+		runOpts := &dockertest.RunOptions{
+			Name:      val.instanceName(),
+			NetworkID: s.dkrNet.Network.ID,
+			Mounts: []string{
+				fmt.Sprintf("%s/:/root/.gaia", val.configDir()),
+			},
+			Repository: "TODO",
```

Comment:
this is where we need to add the docker location?
what dockerfile needs to be built?